blog/2019-02-07-KIEServerOptaPlannerTaskAssignment.adoc

```diff
@@ -0,0 +1,215 @@
+= KIE Server OptaPlanner Task Assignment
+:page-interpolate: true
+:awestruct-author: MusaTalluzi
+:awestruct-layout: blogPostBase
+:awestruct-tags: [useCase, task assigning, kie-server]
+
+https://docs.optaplanner.org/7.17.0.Final/optaplanner-wb-es-docs/html_single/#_ch.kie.server[KIE Server]
+is a standalone server component that can be used to instantiate and execute rules and processes.
+In this blog, you will learn how to build an OptaPlanner service that implements continuous and real-time planning to solve the
+Task Assignment problem and deploy it to the KIE Server.
+
+
+== https://github.com/kiegroup/kie-server-optaplanner-task-assignment/tree/master/task-assignment-kjar[Task Assignment KJAR]
+
+https://developers.redhat.com/blog/2018/03/14/what-is-a-kjar/[KJAR] or "Knowledge JAR" is a standard JAR file that has
+some extra files included (at least a `META-INF/kmodule.xml` file). You will build Task Assignment Service as a KJAR.
+
+NOTE: In order to build the KJAR, the project's pom.xml needs to have a `<packagin>kjar</packaging>` entry and contains
+`kie-maven-plugin`.
+
+=== Problem Description
+
+Assign each task to a spot in an employee’s queue. Each task has a duration which is affected by the employee’s affinity
+level with the task’s customer.
+
+Constraints:
+
+- [Hard] Skill: Each task requires one or more skill. The employee must possess all these skills.
+- [Soft 0] Critical tasks: Complete critical tasks first, sooner than major and minor tasks.
+- [Soft 1] Minimize makespan: Reduce the time to complete all tasks.
+- [Soft 2] Major tasks: Complete major tasks as soon as possible, sooner than minor tasks.
+- [Soft 3] Minor tasks: Complete minor tasks as soon as possible.
+
+=== Domain Model
+
+image::taskAssigningClassDiagram.png[link="taskAssigningClassDiagram.png"]
+You can read more on how to
```

Comment:
Use active voice:
Instead of "You can read more on ...", use "Read more on ..."
https://developers.google.com/style/voice